src/bin/am/server.rs

```diff
@@ -32,9 +38,50 @@ pub(crate) async fn start_web_server(
             }),
         )
         .route("/explorer/", get(explorer::handler))
-        .route("/explorer/*path", get(explorer::handler))
-        .route("/prometheus/*path", any(prometheus::handler))
-        .route("/prometheus", any(prometheus::handler));
+        .route("/explorer/*path", get(explorer::handler));
+
+    // Proxy `/prometheus` to the upstream (local) prometheus instance
+    if should_enable_prometheus {
+        app = app
+            .route("/prometheus/*path", any(prometheus::handler))
+            .route("/prometheus", any(prometheus::handler));
+    }
+
+    // NOTE - this will override local prometheus routes if specified
+    if is_proxying_prometheus {
+        let prometheus_upstream_base = Arc::new(prometheus_proxy_url.clone().unwrap());
+
+        // Define a handler that will proxy to an external Prometheus instance
+        let handler = move |mut req: http::Request<Body>| {
+            let upstream_base = prometheus_upstream_base.clone();
+            // 1. Get the path and query from the request, since we need to strip out `/prometheus`
+            let path_and_query = req
+                .uri()
+                .path_and_query()
+                .map(|pq| pq.as_str())
+                .unwrap_or("");
+            if let Some(stripped_path) = path_and_query.strip_prefix("/prometheus") {
+                // 2. Remove the `/prometheus` prefix.
+                let new_path_and_query =
+                    http::uri::PathAndQuery::from_maybe_shared(stripped_path.to_string())
```

Comment:
```suggestion
                    http::uri::PathAndQuery::from_maybe_shared(stripped_path)
```

Reply:
so when i remove this line i get complaints:

<img width="786" alt="image" src="https://github.com/autometrics-dev/am/assets/3201327/2dbca601-355a-48d9-afe3-934e415ab35b">

Reply:
happening at the line 

```rs
            let path_and_query = req
                .uri()
```

Reply:
so i just removed the `.to_string()` to a different line and it resolved the error 🙃